cms/templatetags/cms_tags.py

```diff
@@ -81,7 +90,12 @@ def _get_page_by_untyped_arg(page_lookup, request, site_id):
                     return page
         else:
             pages = get_page_queryset(site, draft=_use_draft(request))
-            return pages.select_related('node').get(**page_lookup)
+            try:
+                return pages.select_related('node').get(**page_lookup)
+            except Page.MultipleObjectsReturned:
+                messages.warning(request, warning_message)
```

Comment:
I find the `messages` warning inconsistent with the e-mail to admin setup for a non-existing page below.  A log entry in addition to the optional email might be a better solution.